### DISCUSSION TITLE
Skipping IDL to TypeSupport generation

### DISCUSSION THREAD

**Discussion by begleysm** [jmmorato/openddsharp#255] [Q&A]
Is there any easy way to bypass the generation of TypeSupport from IDL and just use an existing TypeSupport.cs file?

I want to be able to build and run my C# GUI in an environment that doesn't actually have/support OpenDDS (the OpenDDS communications won't be used).  I just want to be able to build without OpenDDSharp wanting to use all of the DDS/ACE/TAO infrastructure to build the TypeSupport.cs file.

**Comment by jmmorato** [comment]
I think that's more a solution project structure question than OpenDDSharp library, but here my advice:

- Create a new solution configuration for the environment that does not support OpenDDS
- Add a project constant for that configuration in order to be checked by the preprocessor
- Uncheck the projects not required to build
- Ensure that `TypeSupport.cs` (copied from another build for example) 
- Wrap the OpenDDSharp specific code with `#if...#endif` blocks in your GUI application

Could that work for you?

**Reply by begleysm** [reply]
That makes sense.  It does sound like it would work.  Thanks, we'll try that.
